### PR TITLE
Reduce code crowding in core.py & introduce interactive setup

### PR DESCRIPTION
**Pull request by abhisharma404**
#### Changes this pull request makes :

This PR adds `args_helper.py` module, which handles all the argument parsing task, hence `core.py` is less crowded.

**Features added** :
1. User can now setup their credentials in an interactive manner, i.e. a wizard setup mode.
For example : `sudo python SecureTea.py --twitter` will start an interactive mode, where users need to input their token details one after the other.
```
Usage : 
--twitter       Setup twitter credentials
--twilio_sms    Setup twilio credentials 
--telegram      Setup telegram credentials
```
2. User can also setup all the notification medium at one go, by using `--all` argument.
Example : `sudo python SecureTea.py --all`
3. Earlier Twitter was compulsory, now no particular medium is compulsory, user can abide by any medium.

#### Checklist :

  - [x] I have added the relevant documentation.
  - [x] My branch is up-to-date with the upstream master branch.
  - [x] I have fully tested the changes locally.
  - [x] Supports Python 2 & 3
  - [x] PEP-8 formatted
  - [x] Commit message according to project guidelines

#### Development environment

    OS: Ubuntu 18.04 LTS
    Python Version: 3.6.5

### CODE REVIEW

**Comment by rejahrehim** [comment]
Can we trigger all option if no option provided?

**Comment by abhisharma404** [comment]
@rejahrehim Currently no, but sure I can work on that.

**Comment by rejahrehim** [comment]
@abhisharma404 Okay, It would be better to remove "--all"

**Comment by abhisharma404** [comment]
@rejahrehim Sure :)

**Comment by abhisharma404** [comment]
@rejahrehim I have updated the PR as said.

**Changes** : 
1. '--all' argument has been removed, instead all the integration's setup wizard will run if no argument is passed.
Example : `sudo python SecureTea.py`, this will initiate the interactive setup of all the integration.
2. Skip feature has been added.
If while going through interactive setup, user decides to skip, they can do so by entering `s` or `S`. The program will skip the current integration and jump to next if any.

File: securetea/core.py [hunk 1]
```diff
@@ -37,92 +38,70 @@ class SecureTea(object):
     def __init__(self):
         """Docstring."""
```
Comment (rejahrehim) [inline securetea/core.py]: It would be better to add proper Docstrings.

**Comment by abhisharma404** [comment]
@rejahrehim Codacy is showing an issue, related to function call on Line 80 of `args_helper.py`. Honestly speaking I can't really understand why is it showing it as non callable. Any suggestions?

**Comment by rejahrehim** [comment]
Seems like lots of code repetition in `core.py`. It would be better to make this common. for example multiple initialization messages.

**Comment by abhisharma404** [comment]
> Seems like lots of code repetition in `core.py`. It would be better to make this common. for example multiple initialization messages.

@rejahrehim 
I did'nt get you, should I close this PR and work on `common.py` instead?

**Comment by abhisharma404** [comment]
@rejahrehim I have rebased my PR & made the appropriate changes. Please have a look.

File: securetea/args_helper.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        -------
+        None
+
+        Returns:
+        --------
+        dict: User entered values
+        """
+        def inner_wrapper(*args, **kwargs):
+            skip = False
+            config_dict = func(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (abhisharma404) [reply securetea/args_helper.py]: @rejahrehim Please have a look at this. I can't understand the reason of it saying so.

**Pull request merged**